pkg/autodetect/main.go

```diff
@@ -254,6 +288,13 @@ func (b *Background) cleanDeployments() {
 	}
 
 	jaegerOpts := []client.ListOption{}
+
+	viperWatchNs := viper.GetString("watch-ns")
+	if viperWatchNs != "" {
+		jaegerOpts = append(jaegerOpts, client.InNamespace(viperWatchNs))
```

Comment:
This might be tricky now :-) If my suspicion that the "WATCH_NAMESPACE" can return a list of namespaces, then this function will need to be called once per namespace. Perhaps change this function to accept a namespace as parameter, and create a `cleanDeploymentsWatchedNamespaces()` that iterates over the list and calls `cleanDeployments(namespace string)`?

Reply:
my understanding is that WATCH_NAMESPACE (https://github.com/jaegertracing/jaeger-operator/blob/90b4053280fb230f99da4a6be5b40091d754ae43/deploy/operator.yaml#L27) is the namespace for jaeger, and it could only be one? Does that field accept a list of strings? If Jaeger and Jaeger operator sit in two different namespaces, jaeger operator would need cluster role.

Reply:
POD_NAMESPACE is the namespace where the Jaeger Operator is running. WATCH_NAMESPACE is which namespaces the Jaeger Operator is watching for new events (new/updated/deleted Jaeger CRs). I'm sure multiple values are possible, just not sure what's the format.

Reply:
I saw some operators where WATCH_NAMESPACE can be one namespace or a comma-separated list. I think is ok if we want to support multiple namespaces here.

Reply:
If I'm not wrong, Is not necessary to have ClusterRole permissions to watch multiple namespaces, you only need the appropriate permissions on each namespace.  Of course watching more than one namespace complicates this PR a little bit. so I'm ok if for now we only support one namespace.

Reply:
I just tried and it doesn't seem to be possible to have more than one namespace there. ClusterRole *is* necessary when `WATCH_NAMESPACE` is empty, though.

Reply:
@jpkrohling  Can you please provide me necessary ClusterRole when WATCH_NAMESPACE is empty ?